esmvalcore/_data_finder.py

```diff
@@ -35,15 +36,17 @@ def find_files(dirnames, filenames):
 def get_start_end_year(filename):
     """Get the start and end year from a file name.
 
+    Additionally, returns the start and end dates as strings.
```

Comment:
Instead of adding this new feature to this already complicated function, would it be possible to create a new function `get_start_end_date` that returns this information? If there is code that is shared with the `get_start_end_year` function, that could be moved to another function that is called by both?